### PR TITLE
LSIF: Swagger API documentation

### PR DESCRIPTION
**Pull request by efritz**
Move API documentation into a Swagger document. This has a few benefits:

- is structured (not plaintext, has a specification)
- contains json schema for request/response bodies (can be machine checked)
- is runnable via Swagger UI (instructions included in api.md)

This required adding CORS to the server API so that it can be reached from the Swagger UI running via docker. I don't think this has any ill-effects as this API is not exposed outside of development environments.

### CODE REVIEW

File: lsif/docs/api.md [hunk 1]
```diff
@@ -1,72 +1,15 @@
...
+The LSIF server endpoints are documented as an [OpenAPI v3](https://swagger.io/docs/specification/about/) document [api.yaml](./api.yaml). This document can be viewed locally via docker, as follows.
 
-## LSIF Endpoints
+```bash
+docker run \
+  -e SWAGGER_JSON=/data/api.yaml \
+  -p 8080:8080 \
+  -v `pwd`:/data \
+  swaggerapi/swagger-ui
+```
```
Comment (chrismwendt) [inline lsif/docs/api.md]: I didn't know this had to be run from the `lsif/docs/` directory, which resulted in me only seeing the sample pet store and not knowing what I did wrong.

**Comment by efritz** [comment]
@chrismwendt thanks for taking a look. Do you like the change overall? Is there a different standard that you've seen/prefer?

I did this as an organizational tool for myself but thought it might be useful committed.

File: lsif/docs/api.md (hunk 1 above)
Comment (efritz) [reply lsif/docs/api.md]: Sorry, updated!

**Comment by chrismwendt** [comment]
This seems like a lateral change to me. I don't have any other tools in mind.

- Markdown pro: it's convenient to link to plain markdown (however, we don't do that often because this is an internal API)
- Swagger pro: the UI is nicer
- Swagger pro: can make requests from the page (personally I don't see myself using this)
- Swagger con: have to run a command in the right directory to see the docs (throwing this in a `./dev/` script would help, but still it's a manual step)

With significant effort (doubtful it's worth it), I imagine the next logical steps that would be a clear improvement (not lateral) would be:

- Enforce the Swagger spec by running JSON schema checks in lsif-server
- Autogenerate markdown files from the Swagger spec, enforce that it's up to date by running that in CI
- Add a new endpoint that serves the Swagger UI (e.g. localhost:3080/lsif-server/swagger)

**Pull request merged**